### PR TITLE
Fix external DC photo download

### PR DESCRIPTION
**Pull request by Qusonann**
- switched to NonBlockingProxySocket to make it work
- NotEncryptedSocketMessenger truly async 
- use non-blocking socket for registration
- fix username compare error

### CODE REVIEW

File: src/Client/StatusWatcherClient/ContactsKeeper.php [hunk 1]
```diff
@@ -68,14 +75,26 @@ class ContactsKeeper
...
+
+    /**
+     * @param BasicClient    $client
+     * @param DeferredClient $deferredClient
+     * @param Clock|null     $clock
+     * @param ContactUser[]  $startContacts
+     */
+    public function __construct(
+        BasicClient $client,
+        DeferredClient $deferredClient,
```
Comment (Ririconj) [inline src/Client/StatusWatcherClient/ContactsKeeper.php:91]: Passing 2 clients in the same time

File: src/TGConnection/SocketMessenger/NotEncryptedSocketMessenger.php [hunk 2]
```diff
@@ -66,6 +66,11 @@ private function log(string $code, string $message): void
      */
     public function readMessage(): ?AnonymousMessage
     {
+        if (!$this->socket->ready()) {
+            $this->socket->poll();
+
+            return null;
+        }
```
Comment (Ririconj) [inline src/TGConnection/SocketMessenger/NotEncryptedSocketMessenger.php:73]: All socketmessengers must be async

File: src/Client/StatusWatcherClient/ContactsKeeper.php (hunk 1 above)
Comment (Qusonann) [reply src/Client/StatusWatcherClient/ContactsKeeper.php]: Removed, not needed here

File: src/TGConnection/SocketMessenger/NotEncryptedSocketMessenger.php (hunk 2 above)
Comment (Qusonann) [reply src/TGConnection/SocketMessenger/NotEncryptedSocketMessenger.php:75]: updated, now it is async

**Pull request merged**